src/lib/containers/table/table-top/ProgrammaticOptions.tsx

```diff
@@ -1,60 +1,55 @@
+import { QueryBundleRequest } from '../../../utils/synapseTypes'
 import { ProgrammaticInstructionsModal } from '../../ProgrammaticInstructionsModal'
+import { useGetQueryResultBundleWithAsyncStatus } from '../../../utils/hooks/SynapseAPI'
+import { SynapseConstants } from '../../../utils'
 
-type ProgrammaticOptionsProps = {
+export type ProgrammaticOptionsProps = {
   queryBundleRequest: QueryBundleRequest
-  queryResultBundle: QueryResultBundle
   onHide: () => void
 }
 
 function ProgrammaticOptions({
   queryBundleRequest,
-  queryResultBundle,
   onHide,
 }: ProgrammaticOptionsProps) {
-  const [generatedSql, setGeneratedSql] = useState('')
-  useDeepCompareEffect(() => {
-    const getData = async () => {
-      const { query } = queryBundleRequest
-      const { sql, selectedFacets = [] } = query
-      const { columnModels } = queryResultBundle
-      if (!columnModels) {
-        console.error(
-          'Column Models must be included to complete transform sql request',
-        )
-      }
-      const transformSqlWithFacetsRequest: TransformSqlWithFacetsRequest = {
-        concreteType:
-          'org.sagebionetworks.repo.model.table.TransformSqlWithFacetsRequest',
-        sqlToTransform: sql,
-        selectedFacets,
-        schema: columnModels!,
-      }
+  const [generatedSql, setGeneratedSql] = useState('...')
 
-      try {
-        const res = await getTransformSqlWithFacetsRequest(
-          transformSqlWithFacetsRequest,
-        )
-        // SWC-5686: The ID column is required by the client, and this column may not have been selected!
-        // Change the SQL to "SELECT * ..."
-        const indexOfFrom = res.transformedSql.toUpperCase().indexOf('FROM SYN')
-        const selectStarTransformedSql = `SELECT * ${res.transformedSql.substring(
-          indexOfFrom,
-        )}`
-        setGeneratedSql(selectStarTransformedSql)
-      } catch (e) {
-        console.error('Error on getTransformSqlWithFacetsRequest ', e)
-      }
+  const {
+    data: asyncJobStatus,
+    isLoading: queryIsLoading,
+    error,
+    isPreviousData: newQueryIsFetching,
+  } = useGetQueryResultBundleWithAsyncStatus(
+    {
+      ...queryBundleRequest,
+      partMask: SynapseConstants.BUNDLE_MASK_COMBINED_SQL,
+    },
+    {
+      keepPreviousData: true,
+    },
+  )
+
+  const combinedSql = asyncJobStatus?.responseBody?.combinedSql
+  const isLoadingNewBundle = queryIsLoading || newQueryIsFetching
+
+  useEffect(() => {
+    if (combinedSql && !isLoadingNewBundle) {
+      // SWC-5686: The ID column is required by the client, and this column may not have been selected!
+      // Change the SQL to "SELECT * ..."
+      const indexOfFrom = combinedSql.toUpperCase().indexOf('FROM SYN')
+      const selectStarTransformedSql = `SELECT * ${combinedSql.substring(
+        indexOfFrom,
+      )}`
+      setGeneratedSql(selectStarTransformedSql)
+    }
+  }, [combinedSql, isLoadingNewBundle])
+
+  useEffect(() => {
```

Comment:
Adding `useErrorBoundary` above will throw the error. Then you can delete this effect. Unless there's a case where there is an error only when `isLoadingNewBundle` is true?